daemon/build.go

```diff
@@ -75,15 +76,24 @@ func (rl *releaseableLayer) Release() error {
 	if rl.released {
 		return nil
 	}
+	if err := rl.releaseRWLayer(); err != nil {
+		return err
+	}
+	if err := rl.releaseROLayer(); err != nil {
```

Comment:
Good call.  I also added a little extra logging to `ReleaseROLayer` to make sure even in the case where the code is already processing an error still gets log info if the best effort release of the RO layer fails too.